app/controllers/subscriptions_controller.rb

```diff
@@ -28,6 +28,15 @@ def create
     render json: { id: subscription.id }, status: status
   end
 
+  def change_frequency
+    subscription.update!(frequency: frequency)
+    render json: { subscription: subscription }
+  rescue ArgumentError
+    # This happens if a frequency is provided that isn't included
+    # in the enum which is in the Subscription model
+    raise ActiveRecord::RecordInvalid
```

Comment:
I'm not to fussed about handling the error in this way, it's the best we can do given that enums aren't 100% supported in terms of validation.